test/IDE/complete_macros_expanded.swift

```diff
@@ -0,0 +1,79 @@
+// REQUIRES: swift_swift_parser
+
+// RUN: %empty-directory(%t)
+// RUN: mkdir -p %t/plugins
+
+//##-- Prepare the macro plugin.
+// RUN: %host-build-swift -swift-version 5 -emit-library -o %t/plugins/%target-library-name(MacroDefinition) -module-name=MacroDefinition %S/../Macros/Inputs/syntax_macro_definitions.swift -g -no-toolchain-stdlib-rpath
+
+// RUN: %target-swift-ide-test -batch-code-completion -source-filename %s -filecheck %raw-FileCheck -completion-output-dir %t -plugin-path %t/plugins -parse-as-library 
+
```

Comment:
Since `arbitrary` goes down fairly different paths to named, possibly worth adding tests for it as well?

Reply:
Apparently `arbitrary` at top-level is not really supported:
If we add `names: arbitrary`
* `error: 'declaration' macros are not allowed to introduce arbitrary names at global scope`
* `error: 'peer' macros are not allowed to introduce arbitrary names at global scope`
And the macro decl becomes `isInvalid()`, and the expansion is not performed.

If we omit `names` from the macro decl:
* `error: declaration name '<name>' is not covered by macro '<macro name>'`
Introduced values are found and type checked, but since it's `error`, it doesn't built.

I feel like we should remove 
```cpp
   unexpandedDecls.append(TopLevelArbitraryMacros.begin(),
                          TopLevelArbitraryMacros.end());
```
From `lookupVisibleDecls()`

Reply:
I was thinking within a struct

Reply:
For non top level cases, there is not much different code paths between arbitrary and named, but sure.